sdk/lib/atl/atlwin.h

```diff
@@ -28,6 +28,10 @@
 #define Unused(x)    (x);
 #endif // __GNUC__
 
+#ifndef _STRSAFE_H_INCLUDED_
+    #include <shlwapi.h>
+#endif
```

Comment:
I don't see the relation between strsafe and shlwapi?

Reply:
The below code uses `StringCchPrintf` or `wnsprintf`.

Reply:
`shlwapi.h` is a very heavy header to include for that case,
I'd assume that atl either has it's own functions for this or that it includes strsafe in that case.

Reply:
Used `_sntprintf`. Done.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/snprintf-snprintf-snprintf-l-snwprintf-snwprintf-l?view=msvc-160

Reply:
https://jira.reactos.org/browse/CORE-17503
`<stdio.h>` needs a fix for secure `sprintf/swprintf` functions.

Reply:
And what's the problem in using strsafe.h in ATL?

Reply:
OK, I will cut off the compilers that don't support strsafe.h.

Static CRT library or shlwapi might be good idea for something.

Reply:
It's not a "cut off compilers not supporting strsafe.h", since this header is independent of any compiler and implements its functionality within itself. And we have that header in our code base.